packages/hash/subgraph/src/shared/type-system-patch.ts

```diff
@@ -15,12 +15,24 @@ import {
 
 const versionedUriGroupsPattern = /(.+\/)v\/(\d+)(.*)/;
 
+export const splitVersionedUri = (id: VersionedUri): [BaseUri, number] => {
+  const [baseUri, version] = id.match(versionedUriGroupsPattern)!;
+  return [baseUri, Number(version)];
```

Comment:
This indexing seems wrong, we would have to have an empty placeholder first to select the capture groups
![image](https://user-images.githubusercontent.com/6988668/204795996-0920268f-432b-4a69-8fe3-36721a991332.png)
```suggestion
  const [_, baseUri, version] = id.match(versionedUriGroupsPattern)!;
  return [baseUri, Number(version)];
```

Reply:
Also I am getting a TSC error for `Type 'string | undefined' is not assignable to type 'string'.` with `baseUri`. We may need an assertion `!`